diff --git a/src/rc_compat.c b/src/rc_compat.c
index 0ab9e4d..903c23c 100644
--- a/src/rc_compat.c
+++ b/src/rc_compat.c
@@ -1 +1 @@
-#if !defined(RC_NO_THREADS) && !defined(_WIN32) && !defined(GEKKO) && !defined(_3DS) && (!defined(_XOPEN_SOURCE) || (_XOPEN_SOURCE - 0) < 500)
+#if !defined(RC_NO_THREADS) && !defined(_WIN32) && !defined(GEKKO) && !defined(__3DS__) && (!defined(_XOPEN_SOURCE) || (_XOPEN_SOURCE - 0) < 500)
@@ -197 +197 @@ void rc_mutex_unlock(rc_mutex_t* mutex)
-#elif defined(_3DS)
+#elif defined(__3DS__)
diff --git a/src/rc_compat.h b/src/rc_compat.h
index 3f24df2..f90ee5f 100644
--- a/src/rc_compat.h
+++ b/src/rc_compat.h
@@ -102 +102 @@ RC_BEGIN_C_DECLS
- #elif defined(_3DS)
+ #elif defined(__3DS__)
